app/javascript/src/components/editor/ExpandableCodeblock.svelte

```diff
@@ -0,0 +1,66 @@
+<script>
+  import Expand from "../icon/Expand.svelte"
+
+  export let fullContentLines
+  export let snippetLineCount = 3
+  export let snippetHighlightedLineIndex = 0
+
+  let expanded = false
+  let snippetLines = []
+  $: {
+    snippetLines = []
+
+    const snippetStartLineIdx = snippetHighlightedLineIndex + Math.max(0, Math.round(snippetLineCount / 2))
+    for (let i = snippetStartLineIdx; i < snippetStartLineIdx + snippetLineCount; i++) {
+      snippetLines.push(fullContentLines[i])
+    }
+  }
+
+  function escapeLine(line) {
+    return line.replaceAll(/\s/g, "&nbsp;").replaceAll(/\t/g, "&nbsp;&nbsp;")
+  }
+</script>
+
+<style>
+  .highlighted {
+    background-color: rgba(255, 255, 0, 0.2);
+  }
+
+  .microlight {
+    width: 100%;
+  }
+</style>
+
+<div class="relative">
+  <div class="absolute inline-block cursor-pointer top-0 right-0 p-1/16" on:click={() => expanded = true}>
```

Comment:
It might be nicer to make this into a proper class. We tend to stick to utility classes purely as utilities, rather than primary styling (unless we get lazy)